### PR TITLE
Minor cleanup of Kubectl's downloadKubectl and writeInitScripts

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

### CODE REVIEW

**Review by aleksfront — APPROVED** [review]

File: src/main/kubectl.ts [hunk 1]
```diff
@@ -309,99 +311,86 @@ export class Kubectl {
...
-        });
-        resolve();
-      });
-      stream.pipe(file);
-    });
+    const downloadStream = got.stream({ url: this.url, decompress: true });
+    const fileWriteStream = fs.createWriteStream(this.path, { mode: 0o755 });
+    const pipeline = promisify(stream.pipeline);
+
+    await pipeline(downloadStream, fileWriteStream);
```
Comment (jim-docker) [inline src/main/kubectl.ts:318]: `try`/`catch`? Errors are still possible?

Comment (Nokel81) [reply src/main/kubectl.ts:318]: `promisify` above handles the errors and turns it into rejections

**Review by ixrock — APPROVED** [review]

File: src/main/kubectl.ts (hunk 1 above)
Comment (jim-docker) [reply src/main/kubectl.ts:318]: So `'this.path` gets deleted in the event of an error?

Comment (Nokel81) [reply src/main/kubectl.ts:318]: Oh I see...

**Pull request merged**